docs/api/remix.md

```diff
@@ -467,6 +467,31 @@ See also:
 - [`action`][action]
 - [`useTransition`][usetransition]
 
+### `useRouteData`
+
+This hook returns the JSON parsed data from the route loader function of any matching route in the current URL.
+
+```tsx filename=app/routes/invoices.tsx
+export function loader() {
+  return fakeDb.invoices.findAll();
```

Comment:
`action` & `loader` need to return `Response` or use `json` helper

```suggestion
  return json(await fakeDb.invoices.findAll());
```

Reply:
I'm not 100% sold on this requirement. I know we already merged some docs changes that said we have to do this, but I'd like to find a way to still allow people to return whatever data they want and we just encode/decode for them.

Reply:
I think a lot of the confusion about typing loader return value and useLoaderData have to do with Remix allowing you to return a naked object from the loader.

I strongly recommend deprecating this and having them return an actual Response. That way there is no magic that leads to misunderstanding. They will know that loaders always return a Response.